### PR TITLE
[Verif][LTL] Add basic python binding support

### PR DESCRIPTION
**Pull request by uenoku**
This adds basic python binding support for verif and LTL dialects so that we can parse and analyses output MLIR file with these dialects.

### CODE REVIEW

**Review by fabianschuiki — APPROVED** [review]
Not an expert on the Python bindings, but LGTM when comparing to sibling code :+1:

File: integration_test/Bindings/Python/dialects/verif.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+with Context() as ctx, Location.unknown():
+  circt.register_dialects(ctx)
+  m = Module.create()
+  with InsertionPoint(m.body):
+    i1 = IntegerType.get_signless(1)
+    true = hw.ConstantOp(IntegerAttr.get(i1, 1))
+    assertOp = verif.AssertOp(true)
+    # CHECK: verif.assert %true
+    print(assertOp)
```
Comment (mikeurbach) [inline integration_test/Bindings/Python/dialects/verif.py:17]: nit: github (and apparently some humans) seem to care about the trailing newline at the end of file 🤷

**Review by mikeurbach — APPROVED** [review]
Looks good, thanks for adding it!

File: include/circt-c/Dialect/LTL.h [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+#ifndef CIRCT_C_DIALECT_LTL_H
+#define CIRCT_C_DIALECT_LTL_H
+
+#include "mlir-c/IR.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+MLIR_DECLARE_CAPI_DIALECT_REGISTRATION(Ltl, ltl);
```
Comment (uenoku) [inline include/circt-c/Dialect/LTL.h:14]:
```suggestion
MLIR_DECLARE_CAPI_DIALECT_REGISTRATION(LTL, ltl);
```

File: lib/CAPI/Dialect/LTL.cpp [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "circt-c/Dialect/LTL.h"
+#include "circt/Dialect/LTL/LTLDialect.h"
+
+#include "mlir/CAPI/Registration.h"
+
+MLIR_DEFINE_CAPI_DIALECT_REGISTRATION(Ltl, ltl, circt::ltl::LTLDialect)
```
Comment (uenoku) [inline lib/CAPI/Dialect/LTL.cpp:14]:
```suggestion
MLIR_DEFINE_CAPI_DIALECT_REGISTRATION(LTL, ltl, circt::ltl::LTLDialect)
```

**Pull request merged**